src/backend/distributed/operations/shard_cleaner.c

```diff
@@ -91,32 +92,46 @@ TryDropMarkedShards(bool waitForCleanupLock)
  * will be removed at a later time when there are no locks held anymore on
  * those placements.
  *
+ * If waitForLocks is false, then if we cannot take a lock on pg_dist_placement
+ * we continue without waiting.
+ *
  * Before doing any of this it will take an exclusive PlacementCleanup lock.
  * This is to ensure that this function is not being run concurrently.
  * Otherwise really bad race conditions are possible, such as removing all
- * placements of a shard. waitForCleanupLock indicates if this function should
+ * placements of a shard. waitForLocks indicates if this function should
  * wait for this lock or error out.
```

Comment:
This sentence also seems untrue. If waitForLocks is `false` we simply return `0`, we don't error out.